cdap-proto/src/main/java/io/cdap/cdap/proto/id/ProgramId.java

```diff
@@ -157,4 +146,12 @@ public Iterable<String> toIdParts() {
   public static ProgramId fromString(String string) {
     return EntityId.fromString(string, ProgramId.class);
   }
+
+  /**
+   * Schedule is versionless. Convert versionId in ProgramId to be default version, so it represents
+   * the latest version program
+   */
+  public ProgramId toDefaultVersion() {
```

Comment:
is default : SNAPSHOT?

Reply:
yes

Reply:
Do you think this solution is sustainable? what if we decide to map to latest in future instead of -SNAPSHOT, since everything as of today points to latest?

Reply:
For sustainable solution I think we probably want to bind `ProgramReference` with schedules, but that's not an easy change since ProgramId is already part of the table. This solution is also to accommodate existing schedules from 6.7 created with "-SNAPSHOT" version.  
Even today "-SNAPSHOT" is used to reference "latest" in other places like program LC handlers

Reply:
I see your point, however my concern is that we are not being consistent with our approaches - like in some places we map to latest and some places (like here in schedules) we map to snapshot. Also, going forward we don't want to have this default snapshot I think, so we might run into loopholes. Maybe we should have a discussion on this after 6.8?  But yes, for now this solution works well.